src/Microsoft.VisualStudio.Editor.Razor/DefaultEditorSettingsManagerInternal.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.CodeAnalysis.Razor.Editor;
+
+namespace Microsoft.VisualStudio.Editor.Razor
+{
+    internal class DefaultEditorSettingsManagerInternal : EditorSettingsManagerInternal
+    {
+        public override event EventHandler<EditorSettingsChangedEventArgs> Changed;
+
+        private readonly EditorSettingsManager _editorSettingsManager;
+        private readonly ForegroundDispatcher _foregroundDispatcher;
+
+        public DefaultEditorSettingsManagerInternal(ForegroundDispatcher foregroundDispatcher, EditorSettingsManager editorSettingsManager)
+        {
+            if (foregroundDispatcher == null)
+            {
+                throw new ArgumentNullException(nameof(foregroundDispatcher));
+            }
+
+            if (editorSettingsManager == null)
+            {
+                throw new ArgumentNullException(nameof(editorSettingsManager));
+            }
+
+            _foregroundDispatcher = foregroundDispatcher;
+            _editorSettingsManager = editorSettingsManager;
+            _editorSettingsManager.Changed += OnChanged;
```

Comment:
It's probably worth a comment here that we don't unsubscribe, and we're expecting ESM to use weakevents.